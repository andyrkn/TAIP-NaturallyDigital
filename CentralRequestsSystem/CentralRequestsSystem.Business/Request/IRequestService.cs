using CentralRequestsSystem.Business.Request.Models;
using System.Threading.Tasks;

namespace CentralRequestsSystem.Business.Request
{
    public interface IRequestService
    {
        Task AddRequest(AddRequestModel addRequestModel);
    }
}
