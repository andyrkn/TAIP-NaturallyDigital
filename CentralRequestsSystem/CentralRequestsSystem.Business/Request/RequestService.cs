using System;
using System.Threading.Tasks;
using CentralRequestsSystem.Business.Request.Models;
using CentralRequestsSystem.Core;

namespace CentralRequestsSystem.Business.Request
{
    public class RequestService : IRequestService
    {
        private readonly IRequestRepository _requestRepository;
        public Task AddRequest(AddRequestModel addRequestModel)
            => throw new NotImplementedException();
    }
}
