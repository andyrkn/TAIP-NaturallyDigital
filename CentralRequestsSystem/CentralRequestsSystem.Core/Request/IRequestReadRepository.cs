using CentralRequestsSystem.Core.Repository;

namespace CentralRequestsSystem.Core
{
    public interface IRequestReadRepository : IReadRepository<Request>
    {
    }
}
