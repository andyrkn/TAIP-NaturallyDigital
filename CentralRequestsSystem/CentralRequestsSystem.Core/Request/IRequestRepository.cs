using CentralRequestsSystem.Core.Repository;

namespace CentralRequestsSystem.Core
{
    public interface IRequestRepository : IWriteRepository<Request>
    {
    }
}
