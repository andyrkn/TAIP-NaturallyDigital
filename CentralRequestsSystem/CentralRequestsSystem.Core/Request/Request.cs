using CSharpFunctionalExtensions;
using System;

namespace CentralRequestsSystem.Core
{
    public sealed class Request : Entity
    {
        private Request()
        {

        }

        public string UserAdress { get; private set; }
        public string IdentityProviderAdress { get; private set; }
        public DateTime Date { get; private set; }
        public string Payload { get; private set; }

        public static Result<Request> CreateRequest(string userAdress, string identityProviderAdress, DateTime date, string payload)
            => Result.Ok(new Request
                {
                    UserAdress = userAdress,
                    IdentityProviderAdress = identityProviderAdress,
                    Date = date,
                    Payload = payload
                });
    }
}
