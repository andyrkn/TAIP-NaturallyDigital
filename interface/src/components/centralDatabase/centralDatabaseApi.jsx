import axios from 'axios';
import { decodeRequest, encodeRequest } from "../request.model";

const dotenv = require('dotenv');
dotenv.config();
const centralDatabaseAPI = process.env.REACT_APP_CENTRAL_DATABASE_API;

export const getRequest = async (id) => {
    let request = await axios.get(`${centralDatabaseAPI}/Requests/${id}`)
        .then(response => {
            console.log(response.data);
            return decodeRequest(response.data);
        });
    console.log("Raspuns de la server");
    console.log(request);
    return request;
}

export const putRequest = async (id, payload) => {
    return await axios.put(`${centralDatabaseAPI}/Requests/${id}`, { payload: btoa(JSON.stringify(payload)) });
}

export const deleteRequest = async (id) => {
    return await axios.delete(`${centralDatabaseAPI}/Requests/${id}`);
}