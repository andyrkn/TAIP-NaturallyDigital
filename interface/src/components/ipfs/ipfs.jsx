const IPFS = require('ipfs-api');

export const ipfsNode = new IPFS({ host: 'ipfs.infura.io', port: 5001, protocol: 'https' });

export const uploadContent = async (jsonFile) => {
    console.log('Uploading to IPFS');
    const content = Buffer.from(jsonFile);
    const filesAdded = await ipfsNode.files.add(content);
    return filesAdded[0].hash;
}

export const getContent = async (hash) => {
    console.log('Retrieving from IPFS');
    const fileBuffer = await ipfsNode.files.cat(hash);
    return fileBuffer.toString();
}