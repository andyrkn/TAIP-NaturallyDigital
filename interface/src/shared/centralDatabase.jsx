const centralDatabaseAPI = 'http://109.100.27.188:5000/api';

export default centralDatabaseAPI;