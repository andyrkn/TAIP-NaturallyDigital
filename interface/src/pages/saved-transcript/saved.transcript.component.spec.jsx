import React from 'react';
import { assert } from "chai";
import { shallow, configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import renderer from "react-test-renderer";
import ApprovedTranscript from './saved-transcript.component';
import Request from "../../components/request/request.component";
import Response from "../../components/response/response.component";
import { getContent } from '../../components/ipfs/ipfs';
import { deleteRequest } from '../../components/centralDatabase/centralDatabaseApi';
import { Button } from '@material-ui/core';

configure({
    adapter: new Adapter()
});

describe("Saved transcript page", () => {
    let getAccountAddress, getIdentity, match, getContent, deleteIdentity;

    beforeEach(() => {
        getAccountAddress = jest.fn(() => new Promise(() => "account"));
        getIdentity = jest.fn((address, id) => new Promise(() => { }));
        deleteIdentity = jest.fn((address, id) => new Promise(() => "hash"));
        getContent = jest.fn((hash) => new Promise(() => { }));
        match = { params: { id: 1 } };
    })

    it("makes a snapshot", () => {
        const tree = renderer.create(<ApprovedTranscript />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("renders the page components correctly", () => {
        const wrapper = shallow(<ApprovedTranscript match={match} getIdentity={getIdentity} getAccountAddress={getAccountAddress} 
        getContent={getContent}/>)
        const request = wrapper.find(Request);
        const response = wrapper.find(Response);

        assert.lengthOf(request, 1);
        assert.lengthOf(response, 1);
    })

    it("calls delete method on contract when delete button is clicked", () => {
        let wrapper = shallow(<ApprovedTranscript match={match} getIdentity={getIdentity} getAccountAddress={getAccountAddress} 
            getContent={getContent} deleteIdentity={deleteIdentity}/>)

        wrapper.find(Button).simulate("click");

        expect(deleteIdentity.mock.calls.length).toBe(1);
    })
})