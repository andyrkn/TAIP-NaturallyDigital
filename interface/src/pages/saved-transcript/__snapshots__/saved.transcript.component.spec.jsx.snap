// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Saved transcript page makes a snapshot 1`] = `
Array [
  <div
    className="makeStyles-root-3"
  >
    <header
      className="MuiPaper-root MuiPaper-elevation4 MuiAppBar-root MuiAppBar-positionFixed makeStyles-nav-4 MuiAppBar-colorPrimary mui-fixed"
    >
      <div
        className="MuiBox-root MuiBox-root-53 makeStyles-box-8"
      >
        <img
          alt="DSC-Logo"
          className="navbar-logo"
          src="spiderman.png"
        />
        <button
          aria-label="menu"
          className="MuiButtonBase-root MuiIconButton-root makeStyles-menuButton-6 MuiIconButton-colorInherit MuiIconButton-edgeStart"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root makeStyles-icon-9"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
              />
            </svg>
          </span>
        </button>
      </div>
    </header>
  </div>,
  <main
    className="main"
  >
    <div>
      <h2>
        Request
      </h2>
      <div>
        <span
          className="key"
        >
          Institution: 
        </span>
        <span
          className="institution"
        >
          
        </span>
      </div>
      <div>
        <span
          className="key"
        >
          Request type: 
        </span>
        <span
          className="type"
        >
          
        </span>
      </div>
      <div
        id="response"
      >
        <h3>
          Response
        </h3>
      </div>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button-1 MuiButton-containedPrimary"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          Delete
        </span>
      </button>
      <div
        className="error"
      >
        
      </div>
    </div>
  </main>,
]
`;
